install/kubernetes/helm/istio/values.yaml

```diff
@@ -173,6 +177,10 @@ gateways:
     loadBalancerIP: ""
     serviceAnnotations: {}
     type: LoadBalancer #change to NodePort, ClusterIP or LoadBalancer if need be
+
+    # If set to true, the pilot and citadel mtls ports will be exposed on the ingress gateway
+    meshExpansion: false
+
```

Comment:
This is where you should be adding these ports. I suggest creating a separate ingress-gateway-with-mesh-expansion that has same ports as ingress plus additional ports. Same service name as well. 

This gateway should be disabled by default. Then in docs, you simply need to tell people to override the existing ingress gateway service with the new service (no changes to running gateway), add new servers to the gateway spec and so on. This keeps it cleaner and incremental, without having to add these ports to the templated spec.

Reply:
Running 2 gateways is more expensive ( IP and LB cost ). Large users are likely to do that, but for dev and small developers we can't require that. The mesh expansion gateway is disabled by default (false). I am planning to add another ingress-gateway-ILB - this will get a separate internal IP - but since not everyone has access to ILBs the first priority is using a normal gateway.

Note that this change can be done without helm changes - it's using just standard API - but it just makes it easier for a user.